HopsanGUI/Widgets/TextEditorWidget.cpp

```diff
@@ -399,17 +399,47 @@ void TextEditor::keyPressEvent(QKeyEvent* event)
     //Replace tabs with four whitespaces
     if(event->key() == Qt::Key_Backtab)    //Shift+tab, remove spaces to the left until next tab stop (or as many spaces as possible)
     {
-        QString line = this->textCursor().block().text().toLatin1();
-        int pos = textCursor().positionInBlock();
-        while(pos > 0 && line.at(pos-1) == ' ')
+        if(textCursor().anchor() != textCursor().position())    //Selection exists, indent whole selection
         {
-            this->textCursor().deletePreviousChar();
-            line = this->textCursor().block().text().toLatin1();
-            pos = textCursor().positionInBlock();
-            if(pos%2 == 0)
-            {
-                break;
+            auto cursor = textCursor();
+            QString text = cursor.selection().toPlainText();
+            if(text.startsWith(" ")) {
```

Comment:
I dont understand this, why is always two initial spaces removed, OK maybe I understand, its two spaces at the first line in the selection, since it does not start with a new line